File: client/src/app/common/purchase.ts
import { Customer } from "./customer";
import { Member } from "./member";
import { Order } from "./order";
import { OrderItem } from "./order-item";
import { ShippingAddress } from "./shipping-address";

export class Purchase {

    public member!: Member;
    public customer!: Customer;
    public shippingAddress!: ShippingAddress;
    public order!: Order;
    public orderItems!: OrderItem[];

    constructor() { }
}

File: backend/OrderService/src/main/java/com/example/orderservice/dto/request/Purchase.java
package com.example.orderservice.dto.request;

import com.example.orderservice.entity.*;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {
  
  private Member member;
  private Customer customer;
  private ShippingAddress shippingAddress;
  private Order order;
  private Set<OrderItem> orderItems;
  
}
